frontend/src/Calendar.js

```diff
@@ -2,19 +2,108 @@ import React from 'react'
 
+import moment from 'moment'
+import timeGridPlugin from '@fullcalendar/timegrid'
+
+import {fetchUserCalendar} from './script'
+
+// the conversion of one minute to 60000 milliseconds
+const MIN_TO_MS = 60000;
+
 export default class Calendar extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      fetchDone: false,
+      schedulesTimes: [],
+      scheduleIdx: 0     // Current schedule
+    }
+  }
+
+  componentDidMount(){
+    this.getAllSchedules()
+  }
+
+  async getAllSchedules() {
+    const schedules = await fetchUserCalendar()
+    for(const schedule of schedules) {
+      this.getMeetingTimes(schedule)
+    }
+    
+    this.setState({
+      fetchDone: true
+    })
+  }
+
+  getNextSchedule() {
+    if(this.state.scheduleIdx < this.state.schedulesTimes.length - 1) {
+      this.setState({scheduleIdx : this.state.scheduleIdx + 1})
+    }
+  }
+
+  getPrevSchedule() {
+    if(this.state.scheduleIdx > 0) {
+      this.setState({scheduleIdx : this.state.scheduleIdx - 1})
+    }
+  }
+
+  getMeetingTimes(schedule) {
+    let combinedMeetingTimes = []
+
+    for (const course of schedule.courses) {
+      // Lecture Sections
+      const lectureTime = this.getScheduleSections(course.name, course.lectureSection.meetingTimes)
+      // Lab Sections
+      if (course.labSection) {
+        const labTime = this.getScheduleSections(course.name, course.labSection.meetingTimes)
+        combinedMeetingTimes = combinedMeetingTimes.concat(labTime)
+      }
+      combinedMeetingTimes = combinedMeetingTimes.concat(lectureTime)
+    }
+
+    this.setState({
+      schedulesTimes:  [...this.state.schedulesTimes, combinedMeetingTimes]
+    })
+  }
+  
+  getScheduleSections(courseName, meetingTimes) {
+    const lastSunday = this.getLastSunday(Date.now())
+    const courseMeetingTimes = []
+
+    for (const meetingTime of meetingTimes) {
+      const startTime = new Date(lastSunday.getTime() + meetingTime.start*MIN_TO_MS)
+      const endTime = new Date(startTime.getTime() + meetingTime.duration*MIN_TO_MS)
+
+      courseMeetingTimes.push({
+        title: courseName,
+        start: moment(startTime).format("YYYY-MM-DD HH:mm"),
+        end: moment(endTime).format("YYYY-MM-DD HH:mm")
+      })
+    }
+    return courseMeetingTimes
+  }
+
+  getLastSunday(d) {
+    var t = new Date(d);
+    t.setDate(t.getDate() - t.getDay());
+    t.setHours(0,0,0,0);
+    return t;
+  }
+
   render() {
     return (
-     <FullCalendar
-        plugins={[ dayGridPlugin ]}
-        initialView="dayGridMonth"
-        weekends={false}
-        events={[
-            { title: 'event 1', date: '2019-04-01' },
-            { title: 'event 2', date: '2019-04-02' },
-            { title: getScheduleTitles(), date: getScheduleDates()}
-        ]}
-    />
+      <div>
+        <button onClick = {() => {this.getPrevSchedule()}}>Previous Schedule</button>
+        <button onClick = {() => {this.getNextSchedule()}}>Next Schedule</button>
```

Comment:
If you have a function `foo()`, then

```
() => { foo(); }
() => foo()
foo
```

Are all equivalent. You can replace these with, eg. `onClick={this.getNextSchedule}` and it will avoid instantiating an extra unnecessary lambda.

Additionally, please do not use spaces between properties and their values when declaring HTML elements, ie. `onClick={...}` and not `onClick = {...}`

Reply:
I see what you mean. Originally I tried that but was running into [this](https://www.andrewconnell.com/blog/deal-with-undefined-this-react-event-handler-performant-way/) issue. I will also fix the `onClick=` spacing error

Reply:
While this solution works, you should instead consider binding these functions in your component's constructor instead. If you do, then what I've suggested will work just fine and will save you the overhead of instantiating new lambdas.

Reply:
Got it. I binded the functions into the components constructor [here](https://github.com/googleinterns/step78-2020/pull/43/commits/d82d8120e0cfa8debdafaee44bbc573ea79d61ca) and this should save the overhead you mentioned.